FFmpeg 5.1 deprecation warnings

```c
modules/avformat/audio.c:101:44: error: 'channels' is deprecated [-Werror,-Wdeprecated-declarations]
             sh->au.ctx->sample_rate, sh->au.ctx->channels,
                                                  ^
/usr/local/include/libavcodec/avcodec.h:1005:5: note: 'channels' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
modules/avformat/audio.c:155:9: error: 'channel_layout' is deprecated [-Werror,-Wdeprecated-declarations]
                frame.channel_layout =
                      ^
/usr/local/include/libavutil/frame.h:509:5: note: 'channel_layout' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
modules/avformat/audio.c:156:4: error: 'av_get_default_channel_layout' is deprecated [-Werror,-Wdeprecated-declarations]
                        av_get_default_channel_layout(frame.channels);
                        ^
/usr/local/include/libavutil/channel_layout.h:456:1: note: 'av_get_default_channel_layout' has been explicitly marked deprecated here
attribute_deprecated
^
/usr/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
modules/avformat/audio.c:156:40: error: 'channels' is deprecated [-Werror,-Wdeprecated-declarations]
                        av_get_default_channel_layout(frame.channels);
                                                            ^
/usr/local/include/libavutil/frame.h:642:5: note: 'channels' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
modules/avformat/audio.c:158:10: error: 'channels' is deprecated [-Werror,-Wdeprecated-declarations]
                frame2.channels       = st->ausrc_st->prm.ch;
                       ^
/usr/local/include/libavutil/frame.h:642:5: note: 'channels' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
modules/avformat/audio.c:159:10: error: 'channel_layout' is deprecated [-Werror,-Wdeprecated-declarations]
                frame2.channel_layout =
                       ^
/usr/local/include/libavutil/frame.h:509:5: note: 'channel_layout' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
modules/avformat/audio.c:160:4: error: 'av_get_default_channel_layout' is deprecated [-Werror,-Wdeprecated-declarations]
                        av_get_default_channel_layout(st->ausrc_st->prm.ch);
                        ^
/usr/local/include/libavutil/channel_layout.h:456:1: note: 'av_get_default_channel_layout' has been explicitly marked deprecated here
attribute_deprecated
^
/usr/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
modules/avformat/audio.c:173:36: error: 'channels' is deprecated [-Werror,-Wdeprecated-declarations]
                             frame2.nb_samples * frame2.channels,
                                                        ^
/usr/local/include/libavutil/frame.h:642:5: note: 'channels' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
8 errors generated.
```

--- a/modules/avformat/audio.c
+++ b/modules/avformat/audio.c
@@ -97,8 +97,14 @@ int avformat_audio_alloc(struct ausrc_st **stp, const struct ausrc *as,
 
 	avformat_shared_set_audio(st->shared, st);
 
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(59, 37, 100)
+	int channels = sh->au.ctx->ch_layout.nb_channels;
+#else
+	int channels = sh->au.ctx->channels;
+#endif
+
 	info("avformat: audio: converting %u/%u %s -> %u/%u %s\n",
-	     sh->au.ctx->sample_rate, sh->au.ctx->channels,
+	     sh->au.ctx->sample_rate, channels,
 	     av_get_sample_fmt_name(sh->au.ctx->sample_fmt),
 	     prm->srate, prm->ch, aufmt_name(prm->fmt));
 
@@ -151,13 +157,18 @@ void avformat_audio_decode(struct shared *st, AVPacket *pkt)
 
 		const AVRational tb = st->au.time_base;
 		struct auframe af;
+		int channels = st->ausrc_st->prm.ch;
 
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(59, 37, 100)
+		av_channel_layout_default(&frame2.ch_layout, channels);
+#else
 		frame.channel_layout =
 			av_get_default_channel_layout(frame.channels);
 
-		frame2.channels       = st->ausrc_st->prm.ch;
+		frame2.channels       = channels;
 		frame2.channel_layout =
 			av_get_default_channel_layout(st->ausrc_st->prm.ch);
+#endif
 		frame2.sample_rate    = st->ausrc_st->prm.srate;
 		frame2.format         =
 			aufmt_to_avsampleformat(st->ausrc_st->prm.fmt);
@@ -170,7 +181,7 @@ void avformat_audio_decode(struct shared *st, AVPacket *pkt)
 		}
 
 		auframe_init(&af, st->ausrc_st->prm.fmt, frame2.data[0],
-			     frame2.nb_samples * frame2.channels,
+			     frame2.nb_samples * channels,
 			     st->ausrc_st->prm.srate, st->ausrc_st->prm.ch);
 		af.timestamp = frame.pts * AUDIO_TIMEBASE * tb.num / tb.den;
 

